src/resolver.rs

```diff
@@ -1337,48 +1338,58 @@ impl<'a> Report<'a> {
         }
 
         suggestions.sort_by_key(|item| item.rec.diffstat.count);
+        let mut by_criteria = StableMap::new();
```

Comment:
I don't feel like StableMap is the right one here. Presumably you want BTreeMap so that the criteria are sorted by name, and not just "whatever order they happened to be inserted in"? StableMap is mostly just used for the file formats to minimize random churn on load+save.

Reply:
Good point, will fix.